airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -712,7 +712,160 @@ public void testResetCancelsRunningSync() throws Exception {
+    LOGGER.info("Upgraded source connector per-stream state supported version {}.",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_PER_STREAM_STATE_CONNECTOR_VERSION);
+
+    // add new records and run again.
+    final Database sourceDatabase = testHarness.getSourceDatabase();
+    // get contents of source before mutating records.
+    final List<JsonNode> expectedRecords = testHarness.retrieveSourceRecords(sourceDatabase, STREAM_NAME);
+    expectedRecords.add(Jsons.jsonNode(ImmutableMap.builder().put(COLUMN_ID, 6).put(COLUMN_NAME, "geralt").build()));
+    // add a new record
+    sourceDatabase.query(ctx -> ctx.execute("INSERT INTO id_and_name(id, name) VALUES(6, 'geralt')"));
+    // mutate a record that was already synced with out updating its cursor value. if we are actually
+    // full refreshing, this record will appear in the output and cause the test to fail. if we are,
+    // correctly, doing incremental, we will not find this value in the destination.
+    sourceDatabase.query(ctx -> ctx.execute("UPDATE id_and_name SET name='yennefer' WHERE id=2"));
+
+    LOGGER.info("Starting {} sync 2", testInfo.getDisplayName());
+    final JobInfoRead connectionSyncRead2 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead2.getJob());
+    LOGGER.info("state after sync 2: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    testHarness.assertRawDestinationContains(expectedRecords, new SchemaTableNamePair("public", STREAM_NAME));
+
+    // reset back to no data.
+
+    LOGGER.info("Starting {} reset", testInfo.getDisplayName());
+    final JobInfoRead jobInfoRead = apiClient.getConnectionApi().resetConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitWhileJobHasStatus(apiClient.getJobsApi(), jobInfoRead.getJob(),
+        Sets.newHashSet(JobStatus.PENDING, JobStatus.RUNNING, JobStatus.INCOMPLETE, JobStatus.FAILED));
+
+    LOGGER.info("state after reset: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    testHarness.assertRawDestinationContains(Collections.emptyList(), new SchemaTableNamePair("public",
+        STREAM_NAME));
+
+    // sync one more time. verify it is the equivalent of a full refresh.
+    LOGGER.info("Starting {} sync 3", testInfo.getDisplayName());
+    final JobInfoRead connectionSyncRead3 =
+        apiClient.getConnectionApi().syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead3.getJob());
+    LOGGER.info("state after sync 3: {}", apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connectionId)));
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
```

Comment:
This `false` is very confusing here. Does it means that we will only compare with final tables. and not the temporary tables that airbyte creates? I am not 100% sure even after looking at the code.

Reply:
> we will only compare with final tables. and not the temporary tables that airbyte creates

I think this is correct, but I'm not sure why those temporary tables don't exist in the destination during these acceptance tests - the assertSourceAndDestinationDbInSync() method is just retrieving all tables in the destination, so I would expect those `_airbyte_` tables to be included.

It seems like the only test with this set to true is the `testIncrementalDedupeSync` test, so maybe that has something to do with it? 

Maybe @subodh1810 or @ChristopheDuong could provide some clarity here?

Reply:
Just to clarify my initial comment. It was made to point out that this is not easy to understand when I was doing the review. I had to look at the code to understand what it was doing. I don't think that it should be address in this review but like a potential future improvement.

Reply:
I should have tagged it as FAR (For Another Review)